app/src/helpers/utils/fileUploadErrorMessages.ts

```diff
@@ -1,41 +1,106 @@
+    uploadFileErrors: UploadFilesErrors[],
+): UploadFilesErrorBoxMessages {
+    const result: UploadFilesErrorBoxMessages = {};
+
+    uploadFileErrors.forEach((errorItem) => {
+        const { error, filename = '' } = errorItem;
+        const errorMessage = getErrorBoxErrorMessage(errorItem);
+        if (!(error in result)) {
+            result[error] = { filenames: [filename], errorMessage };
+        } else {
+            result[error]?.filenames?.push(filename);
+        }
+    });
+
+    return result;
+}
+
 export type fileUploadErrorMessageType = {
-    message: string;
+    inline: string;
     errorBox: string;
 };
 
-export const fileUploadErrorMessages = {
+type errorMessageType = { [errorType in UPLOAD_FILE_ERROR_TYPE]: fileUploadErrorMessageType };
+export const fileUploadErrorMessages: errorMessageType = {
     noFiles: {
-        message: 'You did not select any file to upload',
+        inline: 'You did not select any file to upload',
         errorBox: 'You did not select any file to upload',
     },
     duplicateFile: {
-        message: 'The file has the same name as another',
+        inline: 'The file has the same name as another',
         errorBox:
             'This file has the same name as another file you selected. Check the name of these file(s) and try again.',
     },
     fileTypeError: {
-        message: 'The file must be in a PDF file format',
+        inline: 'The file must be in a PDF file format',
         errorBox: 'All files should be in a PDF format',
     },
     fileSizeError: {
-        message: 'Please ensure that all files are less than 5GB in size',
+        inline: 'Please ensure that all files are less than 5GB in size',
         errorBox: 'Please ensure that all files are less than 5GB in size',
     },
-    fileNameError: {
-        message: 'Your file has an incorrect filename',
+    generalFileNameError: {
+        inline: 'Your file has an incorrect filename',
         errorBox:
             'Your filename must follow the format [PDFnumber]_Lloyd_George_Record_[Patient Name]_[NHS Number]_[D.O.B].PDF',
     },
     dateOfBirthError: {
-        message: 'This file contains incorrect patient information',
+        inline: 'The patient’s date of birth does not match this filename',
         errorBox: 'The patient’s date of birth does not match this filename',
     },
     patientNameError: {
-        message: 'This file contains incorrect patient information',
+        inline: 'The patient’s name does not match this filename',
         errorBox: 'The patient’s name does not match this filename',
     },
     nhsNumberError: {
-        message: 'This file contains incorrect patient information',
+        inline: 'The patient’s NHS number does not match this filename',
         errorBox: 'The patient’s NHS number does not match this filename',
     },
+    totalFileNumberUnmatchError: {
+        inline: 'The total file numbers does not match with each others',
```

Comment:
This error message doesn't read very clear to me, what edge case will trigger this message?